src/v/kafka/client/direct_consumer/data_queue.h

```diff
@@ -75,11 +87,33 @@ class data_queue {
         chunked_vector<fetched_topic_data> topics;
         size_t total_bytes{0};
     };
+
+    ss::future<> push(entry entry);
+
+    // specifically, does the queue have capacity for one more fetch
+    bool has_capacity() const;
+
+    // wakes the next push if appropriate
+    inline void update_next_push() {
+        if (has_capacity()) {
+            _waiting_to_push.signal();
+        }
+    }
+
+    // wakes the next pop if appropriate
+    inline void update_next_pop() {
```

Comment:
`inline` is implied, but perhaps move the impl of this method into the `.cc` file?